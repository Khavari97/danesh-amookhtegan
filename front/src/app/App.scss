@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


