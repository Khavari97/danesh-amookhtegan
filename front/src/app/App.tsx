import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
