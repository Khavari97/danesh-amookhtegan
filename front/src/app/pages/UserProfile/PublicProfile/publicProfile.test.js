import React from 'react';
import Enzyme, { shallow, configure, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { Provider } from "react-redux";
import App from "./memberShip";
import store from "../../../redux/store"
import sinon from "sinon";
import { render, screen, act } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Button } from '@material-ui/core';
import renderer from "react-test-renderer";


configure({ adapter: new Adapter() })
const setup = (props = {}) => {
    return shallow(
        <Provider store={store}>
            <App {...props} />
        </Provider>
    );
}

const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
};

describe('test public profile component', () => {

    test('render without error', () => {
        const wrapper = setup();
        expect(wrapper).toBeTruthy();
        expect(wrapper.length).toEqual(1);
    })

    test('test title', () => {
        const wrapper = setup();
        let field = findByTestAttr(wrapper, 'personal')
        expect(field).not.toBeUndefined();
        expect(field.length).toBe(0);
        expect(field).toBeTruthy();
    })

    test('props', () => {
        const wrapper = setup();
        wrapper.setProps({
            avatar: "avatar",
            firstName: 'zahra',
            lastName: 'khavari',
            city: 'tehran',
            province: 'tehran',
            description: 'description',
            url: 'string'
        })
        expect(wrapper.length).toBe(1);
        expect(wrapper).toBeTruthy();
        expect(wrapper).toBeDefined();
    })

    it('check props not render without class', () => {
        let wrapper = setup();

        let props = {
            label: 'button', onClick: () => { }
        }

        wrapper.setProps(props)

        expect(wrapper.props()).not.toEqual(props);
        expect(wrapper.props().label).toBeUndefined()

    })

    it('simulate button', () => {
        let wrapper = setup();
        const mockCallBack = sinon.spy();
        let props = {
            label: 'button', onClick: mockCallBack
        }
        wrapper.setProps(props)
        wrapper.update();
        expect(mockCallBack).toHaveProperty('callCount', 0);
    })


});

describe('test public profile states', () => {
    let fields = [
        [{
            title: 'نام',
            name: 'firstName',

        },
        {
            title: 'نام خانوادگی',
            name: 'lastName',

        }
        ],
        {
            title: 'جنسیت',
            name: 'genderDisplay',

        },
        {
            title: 'نام کاربری',
            name: 'username',

        },
        {
            title: 'ایمیل',
            name: 'email',

        },
        [{
            title: 'استان',
            name: 'province',
            value: 'تهران'
        },
        {
            title: 'شهر',
            name: 'city',

        }],

        {
            title: 'دانشگاه',
            name: 'university',

        },
        {
            title: 'شماره تلفن',
            name: 'phoneNumber',

        }
    ]

    let data = {
        avatar: "http://api.planb.markop.ir/media/defaults/default.png",
        city: "تهران",
        description: "دانشجوی دانشگاه خوارزمی ترم 6",
        email: "atena.ganji.ag@gmail.com",
        firstName: "آتنا",
        genderDisplay: "زن",
        lastName: "گنجی",
        phoneNumber: "+989011625458",
        province: "تهران",
        resume: null,
        skills: [{ code: "21", name: "css" }, { code: "8", name: "Git" }, { code: "7", name: "MYSQL" }],
        university: "خوارزمی",
        url: "/users/atena/",
        username: "atena",
    }

    let projects = [
        {
            name: "اساتید",
            description: "ما دنبال بهترین ساختار برای ایجاد یک تیم قوی برای کمک به دانشجویان هستیم.",
            name: "اساتید",
            role: "حذف شده",
            status: { code: "STARTED", label: "شروع شده" },
            url: "/projects/%D8%A7%D8%B3%D8%A7%D8%AA%DB%8C%D8%AF/",
        },
        {
            name: "نوبت دهی",
            description: "رهبری گروهی بزرگ نیاز به یک تیمی فرهیختته دارد ما از شما دعوت به همکاری می کنیم تا بهترین تیم را برای خود داشته باشیم.",
            name: "نوبت دهی",
            role: "درخواست داده شده",
            status: { code: "STARTED", label: "شروع شده" },
            url: "/projects/%D9%86%D9%88%D8%A8%D8%AA-%D8%AF%D9%87%DB%8C/",
        }
    ]
    let wrapper
    beforeEach(() => {
        wrapper = mount(
            <BrowserRouter>
                <Provider store={store}>
                    <App />
                </Provider>
            </BrowserRouter>
        );
    });

    test('check states', () => {
        wrapper.setState({
            fields: fields,
            data: data,
            projects: projects
        })

        wrapper.update()

        expect(wrapper.state('data')).toEqual(data);
        expect(wrapper.state('data')).not.toEqual({});
        expect(wrapper.state('projects')).toEqual(projects);
        expect(wrapper.state('projects')).not.toEqual({});

        expect(findByTestAttr(wrapper, 'resume').find(Button).prop('disabled')).toEqual(true);
    })
    test('snapshot', () => {
        const snapshot = renderer.create(wrapper).toJSON();
        expect(snapshot).toMatchSnapshot();
    })

});