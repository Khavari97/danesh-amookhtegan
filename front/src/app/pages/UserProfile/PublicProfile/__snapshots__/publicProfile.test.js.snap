// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test public profile states snapshot 1`] = `
<div>
  <div
    className="MuiGrid-root publicInfo MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h1
        className="MuiTypography-root animate__animated animate__zoomIn MuiTypography-h1"
        style={
          Object {
            "marginBottom": "40px",
          }
        }
      >
         پروفایل  
      </h1>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
        data-test="avater"
        style={
          Object {
            "height": "100px",
            "width": "100px",
          }
        }
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiAvatar-fallback"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
      </div>
    </div>
    <div
      className="MuiGrid-root infoGrid MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root header MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <img
            alt=""
            src="customer.svg"
            style={
              Object {
                "width": "30px",
              }
            }
          />
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            data-test="personal"
          >
            اطلاعات شخصی
          </p>
        </li>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                نام
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-test="firstName"
              >
                
              </p>
            </li>
          </div>
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                نام خانوادگی
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-test="lastName"
              >
                
              </p>
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                جنسیت
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
              >
                
              </p>
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                نام کاربری
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
              >
                
              </p>
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                ایمیل
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
              />
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                استان
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-test="province"
              />
            </li>
          </div>
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                شهر
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
                data-test="city"
              />
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                دانشگاه
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
              />
            </li>
          </div>
        </div>
        <div
          className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
          style={
            Object {
              "display": "flex",
            }
          }
        >
          <div>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="title"
              >
                شماره تلفن
              </p>
            </li>
            <li
              className="MuiListItem-root MuiListItem-gutters"
              disabled={false}
            >
              <p
                className="description"
              >
                
              </p>
            </li>
          </div>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root infoGrid MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root header MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <img
            alt=""
            src="new_topic.svg"
            style={
              Object {
                "width": "30px",
              }
            }
          />
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            data-test="about"
          >
            درباره ی من
          </p>
        </li>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <p
            className="title"
          >
            
          </p>
        </li>
      </div>
    </div>
    <div
      className="MuiGrid-root infoGrid MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root header MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <img
            alt=""
            src="development_skill.svg"
            style={
              Object {
                "width": "30px",
              }
            }
          />
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            مهارت ها
          </p>
        </li>
      </div>
    </div>
    <div
      className="MuiGrid-root infoGrid MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root header MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <img
            alt=""
            src="attach.svg"
            style={
              Object {
                "width": "30px",
              }
            }
          />
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            رزومه ضمیمه شده
          </p>
        </li>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
        >
          <img
            alt=""
            src="pdfFile.svg"
            style={
              Object {
                "marginLeft": "4px",
                "width": "40px",
              }
            }
          />
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
            style={
              Object {
                "wordBreak": "break-all",
              }
            }
          >
            رزومه ای بارگزاری نشده است
          </p>
        </li>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
        style={
          Object {
            "textAlign": "left",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
          data-test="resume"
          disabled={true}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={-1}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            دانلود رزومه 
          </span>
        </button>
      </div>
    </div>
    <div
      className="MuiGrid-root infoGrid MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
    >
      <div
        className="MuiGrid-root header MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters"
          disabled={false}
          style={
            Object {
              "padding": "0",
            }
          }
        >
          <img
            alt=""
            src="projects.svg"
            style={
              Object {
                "width": "30px",
              }
            }
          />
          <p
            className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
          >
            پروژه ها
          </p>
        </li>
      </div>
    </div>
  </div>
</div>
`;
